### PR TITLE
1675 create all variant types for icon button

### PR DESCRIPTION
**Pull request by gd2910**
## Summary of the changes
All variant types added for icon variant of ic-button

## Related issue
#1675

### CODE REVIEW

**Review by GCHQ-Developer-299 — DISMISSED** [review]

File: packages/react/src/component-tests/IcButton/IcButtonTestData.tsx [hunk 1]
```diff
@@ -87,3 +153,107 @@ export const PopoverDropdown = () => {
...
   );
 };
+
+export const IconBtnGroup = ({ props }) => {
+  return (
+    <>
+      <div
+        style={{
+          backgroundColor:
+            props.appearance === "light" || props.appearance === "light"
```
Comment (jd239) [inline packages/react/src/component-tests/IcButton/IcButtonTestData.tsx:163]: This OR statement is checking the same thing

File: packages/react/src/component-tests/IcButton/IcButtonTestData.tsx [hunk 2]
```diff
@@ -87,3 +153,107 @@ export const PopoverDropdown = () => {
...
+              ? "#2c2f34"
+              : "#fff",
+          display: "flex",
+          gap: "0.5rem",
+          margin: "1rem",
+          width: "fit-content",
+          padding:
+            props.appearance === "light" || props.appearance === "light"
```
Comment (jd239) [inline packages/react/src/component-tests/IcButton/IcButtonTestData.tsx:171]: This OR statement is checking the same thing

File: packages/react/src/stories/ic-button.stories.mdx [hunk 3]
```diff
@@ -1008,6 +1008,154 @@ export const PopoverDropdown = () => {
...
 </Canvas>
 
+### Icon variants
+
+export const IconBtnGroup = (props) => {
+  return (
+    <>
+      <h1>{props.title}</h1>
+      <div style={{
+          backgroundColor: props.appearance === "light" || props.appearance === "light"  ? "#2c2f34" : "#fff",
```
Comment (jd239) [inline packages/react/src/stories/ic-button.stories.mdx:1018]: Same OR statement being checked

File: packages/react/src/stories/ic-button.stories.mdx [hunk 4]
```diff
@@ -1008,6 +1008,154 @@ export const PopoverDropdown = () => {
...
+          display: "flex",
+          gap: "0.5rem",
+          marginBottom: "1rem",
+          width: "fit-content",
+          padding: props.appearance === "light" || props.appearance === "light"  ? "6px 10px" : "none",
```
Comment (jd239) [inline packages/react/src/stories/ic-button.stories.mdx:1023]: Same OR checks

File: packages/react/src/stories/ic-button.stories.mdx [hunk 5]
```diff
@@ -1008,6 +1008,154 @@ export const PopoverDropdown = () => {
...
+            <path d="M0 0h24v24H0V0z" fill="none" />
+            <path d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z" />
+          </svg>
+        </IcButton>
+      </div>
+    </>
+  )
+}
+
+export const props = {
```
Comment (jd239) [inline packages/react/src/stories/ic-button.stories.mdx:1078]: Can this not be imported from test data file? Will make it easier to update if required down the line.

**Review by jd239 — CHANGES_REQUESTED** [review]
A few updates. Good to see icon button in different variants.

**Review by jd239 — DISMISSED** [review]

**Review by jd239 — APPROVED** [review]

**Pull request merged**